test/fail_compilation/test16365.d

```diff
@@ -0,0 +1,23 @@
+/* REQUIRED_ARGS:
+   PERMUTE_ARGS:
+   TEST_OUTPUT:
+---
+fail_compilation/test16365.d(20): Error: 'this' reference necessary to take address of member f1 in @safe function main
+fail_compilation/test16365.d(22): Error: cannot implicitly convert expression (&f2) of type void delegate() pure nothrow @nogc @safe to void function() @safe
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=16365
+
+struct S
+{
+    void f1() @safe { }
+}
+
+void main() @safe
+{
+    void function() @safe f;
+    f = &S.f1;
```

Comment:
I don't understand this. How can a nonstatic member function (which has `this` as an implicit first argument) get converted to a plain function that takes no arguments?